### DISCUSSION TITLE
Can innerJoin "equals" instead of "equalsExp" ?

### DISCUSSION THREAD

**Discussion by ddxl123** [simolus3/drift#2048] [Q&A] [answered]
For example: 
```
innerJoin( myTable,  myTable.id.equals( queryId ) );
```

Can it be equivalent to?
```
where( myTable.id.equals( queryId ) );
```

**Comment by simolus3** [comment, marked as answer]
This is impossible to answer without seeing two complete queries to compare. For inner joins, it doesn't make a difference if you put a where clause in the `on` clause of a join or in the `where` clause of the whole query. To determine whether two queries are equivalent, you need to check all joins and the where clause.
